app/src/main/java/sdp/moneyrun/menu/JoinGameImplementation.java

```diff
@@ -0,0 +1,230 @@
+        // create game name display
+        createGameNameInfoDisplay(gameRepresentation, gameRow);
+
+        // create player count display
+        createPlayerCountNameInfoDisplay(gameRepresentation, gameRow);
+
+        gameLayout.addView(gameRow, gameParams);
+    }
+
+    @SuppressLint("MissingPermission")
+    private void createJoinButton(Button button, int buttonId, GameRepresentation gameRepresentation){
+        // create join button
+        button.setId(buttonId);
+        button.setText(activity.getString(R.string.join_game_message));
+        button.setOnClickListener(v -> joinLobbyFromJoinButton(gameRepresentation));
+
+        // Modify button if the game is full
+        if (gameRepresentation.getPlayerCount() >= gameRepresentation.getMaxPlayerCount()) {
+            button.setEnabled(false);
+            button.setText(activity.getString(R.string.join_game_full_message));
+        }
+
+        // Modify button if game is too far
+        // Grant permissions if necessary
+        requestLocationPermissions(requestPermissionsLauncher);
+
+        fusedLocationClient.getLastLocation()
+                .addOnSuccessListener(activity, location -> {
+                    // Got last known location. In some rare situations this can be null
+                    // In this case, the game cannot be instanciated
+                    if (location == null) {
+                        Log.e("location", "Error getting location");
+                    }
+                    LocationRepresentation locationRep = new LocationRepresentation(location.getLatitude(), location.getLongitude());
+
+                    double distance = gameRepresentation.getStartLocation().distanceTo(locationRep);
+
+                    if (distance > MAX_DISTANCE_TO_JOIN_GAME) {
+                        button.setEnabled(false);
+                        button.setText(activity.getString(R.string.join_game_too_far_message));
+                    }
+                });
+    }
+
+    private void createGameNameInfoDisplay(GameRepresentation gameRepresentation, TableRow gameRow){
+        TextView nameView = new TextView(activity);
+        String nameText = String.format((activity.getResources().getString(R.string.game_name_display)), gameRepresentation.getName());
+        nameView.setText(nameText);
+        nameView.setPadding(0, 0, 40, 0);
+        gameRow.addView(nameView);
+    }
+
+    private void createPlayerCountNameInfoDisplay(GameRepresentation gameRepresentation, TableRow gameRow){
+        TextView playerNumberView = new TextView(activity);
+        String playerNumberText = String.format((activity.getResources().getString(R.string.game_player_number_display)),
+                gameRepresentation.getPlayerCount(),
+                gameRepresentation.getMaxPlayerCount());
+        playerNumberView.setText(playerNumberText);
+        gameRow.addView(playerNumberView);
+    }
+
+    private void joinLobbyFromJoinButton(GameRepresentation gameRepresentation){
```

Comment:
This button should be tested

Reply:
There may be 0 or more buttons depending on the database, and some of the existing buttons may be unclickable. This is hard to test without database emulators.

Reply:
Fair enough